config/env.config.js

```diff
@@ -55,5 +55,46 @@ const config = {
     }
 };
 
+const RedirectPlugin = function(options) { 
+    this.apply = function(compiler) {
+        compiler.plugin('compilation', function(compilation) {
+            compilation.plugin('html-webpack-plugin-before-html-processing', function(htmlPluginData, callback) {
+                let htmlHead = htmlPluginData.html.match('<head>').index;
+                htmlPluginData.html = htmlPluginData.html.slice(0, htmlHead) + '<head>' +
+                    `<script>
+                        var getParameterByName = function(name) {
+                            var url = window.location.search;
+                            var queryExp = new RegExp("[\\?&]"+name+"=([^&#]*)", "i");
+                            var match = queryExp.exec(url);
+                            if (match && match.length > 1) {
+                                return match[1];
+                            }
+                            return "";
+                        };
+                        var originUrl = decodeURIComponent(getParameterByName("originEnvironment"));
+                        var newUrl = window.localStorage.getItem("environmentConfig");
+                        if (newUrl && !window.location.href.match(newUrl) && originUrl !== newUrl) {
```

Comment:
I believe this is going to create a redirect loop where you have prod pointing to edge, edge pointing to insider, then try to navigate from insider back to prod. Please think through how to solve this issue or test to make sure we do not get in any loop.